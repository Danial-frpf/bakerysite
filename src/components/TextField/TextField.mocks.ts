import { ITextField } from ".";

const base: ITextField = {
    label: "TextArea",
    name: "text",
};

export const mockTextFieldProps = {
    base,
};
