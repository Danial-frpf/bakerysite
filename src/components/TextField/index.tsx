import { useState } from "react";
import Label from "../Label";
import styles from "./styles.module.css";

export interface ITextField {
    label: string;
    name: string;
    rows?: number;
    cols?: number;
    placeholder?: string;
    size?: "sm" | "md" | "lg" | "xl";
    props?: React.DetailedHTMLProps<
        React.TextareaHTMLAttributes<HTMLTextAreaElement>,
        HTMLTextAreaElement
    >;
}
const TextField: React.FC<ITextField> = ({
    label,
    name,
    rows,
    cols,
    placeholder,
    size = "md",
    props,
}) => {
    const [value, setValue] = useState("");

    const changeHandler = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
        setValue(e.currentTarget.value);
    };

    return (
        <div data-testid="TextField" className={styles.container}>
            <Label label={label} size={size} />
            <textarea
                name={name}
                rows={rows}
                cols={cols}
                placeholder={placeholder}
                value={value}
                onChange={changeHandler}
                {...props}
                className={`${styles.textarea} ${props?.className}`}
            ></textarea>
        </div>
    );
};

export default TextField;
