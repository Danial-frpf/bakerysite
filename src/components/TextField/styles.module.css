.container {
    text-align: center;
    width: 100%;
}

.textarea {
    font-size: var(--p);
    width: 100%;
    background-color: var(--secondary-orange-40);
    border: 2px solid var(--main-800);
    padding: 1vh 1vw;
}

.textarea:focus {
    background-color: var(--secondary-yellow-40);
    border-color: var(--secondary-orange);
}
