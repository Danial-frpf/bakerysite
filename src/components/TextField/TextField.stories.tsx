import { ComponentStory, ComponentMeta } from "@storybook/react";
import TextField, { ITextField } from ".";
import { mockTextFieldProps } from "./TextField.mocks";

export default {
    title: "templates/TextField",
    component: TextField,
    // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
    argTypes: {},
} as ComponentMeta<typeof TextField>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof TextField> = (args) => (
    <TextField {...args} />
);

export const Base = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args

Base.args = {
    ...mockTextFieldProps.base,
} as ITextField;
