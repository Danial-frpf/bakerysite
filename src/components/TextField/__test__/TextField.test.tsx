import { createRoot } from "react-dom/client";
import { act } from "react-dom/test-utils";
import TextField from "..";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);
it("renders correctly without crashing", () => {
    const div = document.createElement("div");
    act(() => {
        createRoot(div!).render(
            <TextField label="test" name="test"></TextField>
        );
    });
});

it("renders TextField correctly", () => {
    const { getByTestId } = render(
        <TextField label="test" name="test"></TextField>
    );
    expect(getByTestId("TextField")).toHaveTextContent("test");
});

it("matches snapshot", () => {
    const tree = renderer
        .create(<TextField label="test" name="test"></TextField>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
