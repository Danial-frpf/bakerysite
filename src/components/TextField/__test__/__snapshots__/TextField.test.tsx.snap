// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
  data-testid="TextField"
>
  <h4
    data-testid="Label"
  >
    <span>
      test
    </span>
  </h4>
  <textarea
    className="textarea undefined"
    name="test"
    onChange={[Function]}
    value=""
  />
</div>
`;
